File: code/bit_manipulation/src/binary-graycode-conversion/graycode_conversion.php
<?php
// Binary to Gray
function binaryToGray($n) {
    return $n ^ ($n >> 1);
}

// Gray to Binary
function grayToBinary($n) {
    $binary = $n;
    while ($n > 0) {
        $n >>= 1;
        $binary ^= $n;
    }
    return $binary;
}

$num = 7;
echo "Gray Code: " . binaryToGray($num) . "\n";  // Output: 4
echo "Binary Code: " . grayToBinary(binaryToGray($num)) . "\n"; // Output: 7
?>

File: code/bit_manipulation/src/binary-graycode-conversion/graycode_conversion.java
public class graycode_conversion {
    // Binary to Gray
    public static int binaryToGray(int n) {
        return n ^ (n >> 1);
    }

    // Gray to Binary
    public static int grayToBinary(int n) {
        int binary = n;
        while (n > 0) {
            n >>= 1;
            binary ^= n;
        }
        return binary;
    }

    public static void main(String[] args) {
        int num = 7;
        int gray = binaryToGray(num);
        System.out.println("Gray Code: " + gray); // Output: 4
        System.out.println("Binary Code: " + grayToBinary(gray)); // Output: 7
    }
}
